lib/html_parser.dart

```diff
@@ -703,10 +711,12 @@ class ContainerSpan extends StatelessWidget {
 class StyledText extends StatelessWidget {
   final InlineSpan textSpan;
   final Style style;
+  final double textScaleFactor;
 
   const StyledText({
     this.textSpan,
     this.style,
+    this.textScaleFactor = 1.0,
```

Comment:
Is there a reason to default this and make it a property of `StyledText`? Are we exposing it somewhere else?

Reply:
The underlying problem we have here is that `RichText` (which is what `Rich.text()` ultimately creates) will take the `textScaleFactor` that is passed to it and will rescale based on that. Due to the way that we're constructing the widget tree, we have nested `RichText` objects so if `textScaleFactor` is anything other than `1.0`, then we end up with the scaling getting applied multiple times.

By defaulting to `1.0`, we allow `StyledText` objects to be created all over the place with the correct internal scaling value. And then, on line 74, just before we return the final widget tree in the `build()` function we create one more outermost `StyledText` but this time we pass in the real scaling factor so that everything underneath gets scaled correctly. Hope that makes sense.

Reply:
This makes sense. When I asked I was just doing general code review since I haven't had time to take a look at the specific issue or even run the PR.

After running it, reading what you had to say and actually reading the code, everything looks okay, I don't see any problems with this, I will merge it.